### PR TITLE
Add lua params

### PR DESCRIPTION
**Pull request by p00f**

### CODE REVIEW

**Comment by theHamsta** [comment]
Shouldn't it be
```
(parameters
    (indentifier))
```
(function (identifier)) is the function name and only for exported/global functions and not for local ones.

**Comment by p00f** [comment]
The query file uses the same syntax for function names which confused me too https://github.com/nvim-treesitter/nvim-treesitter/blob/3b8efe7a1ebfd5cc84bc0b43e8e5142eb4ef9de8/queries/lua/highlights.scm#L110.

But then I tried it and it worked - my query captures the parameter names correctly:
image
(Colorscheme highlights parameters in orange italic)

**Comment by theHamsta** [comment]
Please change it to
```
(parameters
    (indentifier))
```
This means, highlight a child of function that's either an identifier (the function name as you showed above) or `parameters`
```
(function [(parameters) (identifier)] @parameter)
```

**Comment by p00f** [comment]
Done

**Comment by p00f** [comment]
But in the case of line 110, `identifier` is a child of `function_name` 
image

Shouldn't line 110 be `(function [(function_name (identifier)) @function]`?

**Comment by theHamsta** [comment]
You can probably remove the `identifier` from this query. `function_name` is enough
```
function [34, 0] - [44, 3]
  function_name [34, 9] - [34, 22]
    function_name_field [34, 9] - [34, 22]
      object: identifier [34, 9] - [34, 10]
      property_identifier [34, 11] - [34, 22]
```

**Comment by p00f** [comment]
Thanks, understood. Should I touch that if I didn't write it?

**Comment by p00f** [comment]
Similarly in line 113:
https://github.com/nvim-treesitter/nvim-treesitter/blob/3b8efe7a1ebfd5cc84bc0b43e8e5142eb4ef9de8/queries/lua/highlights.scm#L113
 `local_function` doesn't seem to have function name:
image

File: queries/lua/highlights.scm [hunk 1]
```diff
@@ -121,6 +121,9 @@
 (function_call (identifier) @function . (arguments))
 (function_call (field_expression (property_identifier) @function) . (arguments))
 
+;; Parameters
+(function [(parameters (identifier))] @parameter)
```
Comment (vigoux) [inline queries/lua/highlights.scm:125]: Why having only one alternative in here?

Comment (theHamsta) [reply queries/lua/highlights.scm:125]: why having `function` here? It's redundant.

Comment (p00f) [reply queries/lua/highlights.scm:125]: > Why having only one alternative in here? 

Wdym ?

Comment (p00f) [reply queries/lua/highlights.scm]: > why having `function` here? It's redundant. Also it does not apply to local_function

Done :+1:

**Review by vigoux — APPROVED** [review]

**Pull request merged**